tools/version-bump/lib/update.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { readFile, writeFile, stat } from 'fs/promises';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from './logger';
+
+/**
+ * Update plugin readme stable tag.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeStableTag = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/Stable tag: \d.\d.\d\n/m,
+			`Stable tag: ${ nextVersion }\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme stable tag' );
+	}
+};
+
+/**
+ * Update plugin readme changelog.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateReadmeChangelog = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/readme.txt`;
+	try {
+		const readmeContents = await readFile( filePath, 'utf8' );
+
+		const updatedReadmeContents = readmeContents.replace(
+			/= \d.\d.\d \d\d\d\d-XX-XX =\n/m,
+			`= ${ nextVersion } ${ new Date().getFullYear() }-XX-XX =\n`
+		);
+
+		await writeFile( filePath, updatedReadmeContents );
+	} catch ( e ) {
+		Logger.error( 'Unable to update readme changelog' );
+	}
+};
+
+/**
+ * Update plugin class file.
+ *
+ * @param  plugin      plugin to update
+ * @param  nextVersion version to bump to
+ */
+export const updateClassPluginFile = async (
+	plugin: string,
+	nextVersion: string
+): Promise< void > => {
+	const filePath = `plugins/${ plugin }/includes/class-${ plugin }.php`;
+
+	try {
+		await stat( filePath );
+	} catch ( e ) {
+		// Class file does not exist, return early.
```

Comment:
If the class file doesn't exist is this an expected error? should this throw?

Reply:
No it should not throw an error. Only core has the `class-woocommerce.php` file with a version requiring update, so I want to return early if it does not exist.

I wanted to do this, but `exist` is [deprecated](https://nodejs.org/api/fs.html#fsexistspath-callback) so I need to use `stat` or `access` both of which return errors when the file does not exist.

```js
if ( exists( filePath ) { 
    return;
}
```